modules/currency-l0/src/main/scala/org/tessellation/currency/l0/snapshot/services/StateChannelSnapshotService.scala

```diff
@@ -79,14 +79,21 @@ object StateChannelSnapshotService {
       } yield binary
 
       def consume(signedArtifact: Signed[CurrencySnapshotArtifact], context: CurrencySnapshotContext): F[Unit] = for {
+        binary <- createBinary(signedArtifact)
+        binaryHashed <- binary.toHashed
+        identifier <- identifierStorage.get
+        _ <- lastBinaryHashStorage.set(binaryHashed.hash)
         _ <- dataApplicationSnapshotAcceptanceManager.traverse { manager =>
           snapshotStorage.head.map { lastSnapshot =>
             lastSnapshot.flatMap { case (value, _) => value.dataApplication }
           }.flatMap(manager.consumeSignedMajorityArtifact(_, signedArtifact))
         }
-        binary <- createBinary(signedArtifact)
-        binaryHashed <- binary.toHashed
-        identifier <- identifierStorage.get
+        _ <- snapshotStorage
+          .prepend(signedArtifact, context.snapshotInfo)
+          .ifM(
+            Applicative[F].unit,
+            logger.error("Cannot save CurrencySnapshot into the storage")
```

Comment:
It's not my current change. Also, the whole snapshot info should not be showed to log I believe. It may be a really big object.
anyway, I've changed it to:
```
logger.error(s"Cannot save CurrencySnapshot ordinal=${signedArtifact.ordinal} for metagraph identifier=${context.address} into the storage.")

```